transport/headers.go

```diff
@@ -0,0 +1,31 @@
+package transport
+
+// Edge request context propagation related headers, as defined in
+// https://pages.github.snooguts.net/reddit/baseplate.spec/component-apis/thrift#edge-request-context-propagation
```

Comment:
we probably should mention in the comment that those headers are only used in thrift and grpc (for http we use different headers, usually with `x-` prefix).